main.py

```diff
@@ -148,7 +152,40 @@ def setup_target_task_training(args, target_tasks, model, strict):
                     args.allow_untrained_encoder_parameters, "No best checkpoint found to evaluate."
                 )
                 log.warning("Using untrained encoder parameters!")
-    return task_names_to_avoid_loading
+
+        # The version of the model before target task training for the new task.
+        pre_target_train_model = copy.deepcopy(model)
+
+        if args.transfer_paradigm == "frozen":
+            # will be empty if elmo = 0. scalar_mix_0 should always be
+            # pretrain scalars
+            elmo_scalars = [
+                (n, p)
+                for n, p in model.named_parameters()
+                if "scalar_mix" in n and "scalar_mix_0" not in n
+            ]
+            # Fails when sep_embs_for_skip is 0 and elmo_scalars has nonzero
+            # length.
+            assert_for_log(
+                not elmo_scalars or args.sep_embs_for_skip,
+                "Error: ELMo scalars loaded and will be updated in do_target_task_training but "
+                "they should not be updated! Check sep_embs_for_skip flag or make an issue.",
+            )
+
+        if args.transfer_paradigm == "finetune":
```

Comment:
Merge this with the above. Should only need one if/else for this whole section of code.